perfkitbenchmarker/linux_virtual_machine.py

```diff
@@ -708,3 +732,137 @@ def SetupProxy(self):
+    self.docker_id = resp[:-1]
+    return self.docker_id
+
+  def RemoteCommand(self, command,
+                    should_log=False, retries=SSH_RETRIES,
+                    ignore_failure=False, login_shell=False,
+                    suppress_warning=False):
+    """Runs a command inside the container.
+
+    Args:
+      command: A valid bash command.
+      should_log: A boolean indicating whether the command result should be
+          logged at the info level. Even if it is false, the results will
+          still be logged at the debug level.
+      retries: The maximum number of times RemoteCommand should retry SSHing
+          when it receives a 255 return code.
+      ignore_failure: Ignore any failure if set to true.
+      login_shell: Run command in a login shell.
+      suppress_warning: Suppress the result logging from IssueCommand when the
+          return code is non-zero.
+
+    Returns:
+      A tuple of stdout and stderr from running the command.
+    """
+    # Escapes bash sequences
+    command = command.replace("'", r"'\''")
+
+    logging.info('Docker running: %s' % command)
+    command = "sudo docker exec %s bash -c '%s'" % (self.docker_id, command)
+    return self.RemoteHostCommand(command, should_log, retries,
+                                  ignore_failure, login_shell, suppress_warning)
+
+  def ContainerCopy(self, file_name, container_path='', copy_to=True):
+    """Copies a file to and from container_path to the host's vm_util.VM_TMP_DIR.
+
+    Args:
+      file_name: Name of the file in the host's vm_util.VM_TMP_DIR.
+      container_path: Optional path of where to copy file on container.
+      copy_to: True to copy to container, False to copy from container.
+    Raises:
+      RemoteExceptionError: If the source container_path is blank.
+    """
+    if copy_to:
+      if container_path == '':
+        container_path = CONTAINER_WORK_DIR
+
+      # Everything in vm_util.VM_TMP_DIR is directly accessible
+      # both in the host and in the container
+      source_path = posixpath.join(CONTAINER_MOUNT_DIR, file_name)
+      command = 'cp %s %s' % (source_path, container_path)
+      self.RemoteCommand(command)
+    else:
+      if container_path == '':
+        raise errors.VirtualMachine.RemoteExceptionError('Cannot copy '
+                                                         'from blank target')
+      destination_path = posixpath.join(CONTAINER_MOUNT_DIR, file_name)
+      command = 'cp %s %s' % (container_path, destination_path)
+      self.RemoteCommand(command)
+
+  def RemoteCopy(self, file_path, remote_path='', copy_to=True):
+    """Copies a file to or from the container in the remote VM.
+
+    Args:
+      file_path: Local path to file.
+      remote_path: Optional path of where to copy file inside the container.
+      copy_to: True to copy to VM, False to copy from VM.
+    """
+    if copy_to:
+      file_name = posixpath.basename(file_path)
```

Comment:
I think this has to be:
    if copy_to:
      base_name = os.path.basename(file_path)
      tmp_path = posixpath.join(vm_util.VM_TMP_DIR, base_name)
      self.RemoteHostCopy(file_path, tmp_path, copy_to)
      self.ContainerCopy(base_name, remote_path, copy_to)
    else:
....